api/pkg/template/config.go

```diff
@@ -277,19 +279,11 @@ func (e *Engine) configValueChanged(itemName string) bool {
 		return true
 	}
 
-	return prevVal.Value != currentVal.Value
-}
-
-func (e *Engine) getItemFilename(configItem *kotsv1beta1.ConfigItem) string {
-	// Priority: user value
-	if userVal, exists := e.configValues[configItem.Name]; exists {
-		return userVal.Filename
+	if prevVal.Filename != currentVal.Filename {
```

Comment:
this makes sure to invalidate the cache if the file name changes, not just the value